src/main/java/net/barrage/school/java/ecatalog/app/ProductServiceImpl.java

```diff
@@ -2,13 +2,20 @@
 
 import net.barrage.school.java.ecatalog.model.Product;
 import org.apache.commons.lang3.StringUtils;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
 
 @Service
+@EnableCaching
```

Comment:
All `Enable*` like annotations are only suitable for configrations. U cant just put to random bean. https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/cache/annotation/EnableCaching.html